### PR TITLE
feat(accordion): #233 add accordion atom

### LINKED ISSUE #233: Add accordions to the dress code
Large amount of information and navigation lists are to be presented to user often and dress code lists do not solve the purpose all the times.

Add support for pure css + html accordions (similar to jquery accordion) without any javascript code.

Something as below:

screen shot 2016-08-02 at 11 42 43

screen shot 2016-08-02 at 11 42 57

### PR DESCRIPTION
**Pull request by ashwinaggarwal**
add accordion atom to dress code
close #233

### CODE REVIEW

**Comment by ashwinaggarwal** [comment]
Demo pictures:

screen shot 2016-08-03 at 16 43 43

screen shot 2016-08-03 at 16 44 08

**Comment by sambhav-gore** [comment]
Hello @ashwinaggarwal thank you for contributing to the Dress Code. 

@gabrielhl Will need your inputs on this design.

**Comment by ashwinaggarwal** [comment]
screen shot 2016-08-05 at 11 48 22

screen shot 2016-08-05 at 11 48 41

File: docs/demo/materials/03-atoms/accordions/01-accordion.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+title: accordion
+notes: |
+    Use accordions to separate content into logical sections.
+
+    `dc-accordion__item` is logically separated into two sections `dc-accordion__header` and `dc-accordion__content`
+---
+
+<div class="dc-accordion">
+	<div class="dc-accordion__item">
+		<label class="dc-accordion__header">
```
Comment (sambhav-gore) [inline docs/demo/materials/03-atoms/accordions/01-accordion.html]: As we do not use inputs anymore, can we change this `label` element to `header` to keep it more symantic ?

File: src/styles/atoms/_accordion.scss [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    padding: $dc-space100 $dc-space150 0;
+}
+
+@mixin dc-accordion__item--is-active {
+    outline: 1px solid $dc-blue40;
+}
+
+
+@mixin dc-accordion__item--is-open(
+    $accordion-header: "dc-accordion__header",
```
Comment (sambhav-gore) [inline src/styles/atoms/_accordion.scss]: We do not need this parameter now.

**Comment by gabrielhl** [comment]
Design comments: arrows should go on the right side, use a simple triangular arrow (same as in the select componeent) instead of the one you used.

**Comment by ashwinaggarwal** [comment]
@gabrielhl not showing arrows anymore.. as we discussed to leave the implementation of icons to user.. should we still need to add them?

**Comment by gabrielhl** [comment]
I would say leave it like this, since it's more generic. Maybe in the future add some variant classes to easily add/remove icons from either side, without hardcoding a specific icon.

File: src/styles/atoms/_accordion.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    &:hover {
+        background-color: $dc-gray80;
+    }
+
+    .#{$accordion-content-selector} {
+        height: 0;
+        padding-right: $dc-space150;
+        padding-left: $dc-space150;
+        overflow: hidden;
+    }
```
Comment (sambhav-gore) [inline src/styles/atoms/_accordion.scss]: This is much simplified now, however, I think these three properties do not change in state open/closed so these can belong in it's own selector mixin named `dc-accordion__content`

**Comment by gregdeane** [comment]
@ashwinaggarwal please update this branch with the `develop` branch and fix any potential merge conflicts.

**Comment by ashwinaggarwal** [comment]
@gregdeane this branch has been updated by merging develop to it

**Comment by gabrielhl** [comment]
@ashwinaggarwal Some comments, please fix and I'll +1 and merge

1 - Text problems in the options demo 
- First line: the word "show" is repeated
- Second line: missing a space after the :


2 - Style & design: 
- make the background in the open and active states `#fff`
- use the same hover color as is used on tables (for consistency)
- add `font-weight:500` to the header 


This makes for a very simple, readable default style that matches other elements (such as tables). that can be easily customized. 

And thanks again for your contributions, it's really great!

**Comment by ashwinaggarwal** [comment]
@gabrielhl all changes incorporated as shown below, sorry for the delay :)

screen shot 2016-09-28 at 13 47 28


@xonic moved the accordions from atoms to molecules

**Comment by rbarilani** [comment]
👍 @xonic ?

**Comment by ashwinaggarwal** [comment]
@rbarilani, me and @xonic made some changes on my machine, I've to change the code a bit for those changes, will push them by EOD.

**Comment by ashwinaggarwal** [comment]
@rbarilani updated the code as per the inputs from @xonic 

screen shot 2016-11-01 at 12 09 40
screen shot 2016-11-01 at 12 09 59

**Comment by ashwinaggarwal** [comment]
@xonic @rbarilani are we still waiting on anything?

**Comment by xonic** [comment]
👍

**Pull request merged**